File: OCP/violations/Payment.cs
namespace OCP.Violations
{
    class PaymentService
    {
        public void Pay(string method, double amount)
        {
            if (method == "CREDIT_CARD")
            {
                Console.WriteLine("Paying with credit card");
            }
            else if (method == "UPI")
            {
                Console.WriteLine("Paying with UPI");
            }
            else if (method == "PAYPAL")
            {
                Console.WriteLine("Paying with PayPal");
            }
        }
    }
}

File: OCP/violations/Payment.ts
class PaymentService {
    pay(method: string, amount: number): void {
        if (method === "CREDIT_CARD") {
            console.log("Paying with credit card");
        } else if (method === "UPI") {
            console.log("Paying with UPI");
        } else if (method === "PAYPAL") {
            console.log("Paying with PayPal");
        }
    }
}
